src/index.ts

```diff
@@ -53,22 +55,37 @@ export class GraphQLServer {
     if (props.schema) {
       this.executableSchema = props.schema
     } else if (props.typeDefs && props.resolvers) {
-      let { directiveResolvers, typeDefs, resolvers } = props
+      let uploadMixin = {}
+      let { directiveResolvers, resolvers, typeDefs } = props
+
+      if (Array.isArray(typeDefs)) {
+        const toSingleTypedef = (previousValue: string, typeDef: ITypeDefinitions ): string => {
+          let currentValue = typeof typeDef === 'function' ? typeDef() : typeDef
+          let accumulator = Array.isArray(currentValue)
+            ? currentValue.reduce(toSingleTypedef, '')
+            : currentValue
+
+          return previousValue + accumulator
+        }
 
-      // read from .graphql file if path provided
-      if (typeDefs.endsWith('graphql')) {
-        const schemaPath = path.resolve(typeDefs)
+        typeDefs = typeDefs.reduce(toSingleTypedef, '')
+      }
+
+      if (typeof typeDefs === 'string') {
```

Comment:
I'm not used to TS, is there a fancier way to do that?
TS was complaining about `typeDefs` could not be a string.